models/tcp_route.go

```diff
@@ -27,11 +27,13 @@ type TcpMappingEntity struct {
 	// We don't add uniqueness on InstanceId so that if a route is attempted to be created with the same detals but
 	// different InstanceId, we fail uniqueness and prevent stale/duplicate routes. If this fails a route, the
 	// TTL on the old record should expire + allow the new route to be created eventually.
-	InstanceId       string `gorm:"null; default:null;" json:"instance_id"`
-	ExternalPort     uint16 `gorm:"not null; unique_index:idx_tcp_route; type: int" json:"port"`
-	ModificationTag  `json:"modification_tag"`
-	TTL              *int   `json:"ttl,omitempty"`
-	IsolationSegment string `json:"isolation_segment"`
+	InstanceId           string `gorm:"null; default:null;" json:"instance_id"`
+	ExternalPort         uint16 `gorm:"not null; unique_index:idx_tcp_route; type: int" json:"port"`
+	ModificationTag      `json:"modification_tag"`
+	TTL                  *int   `json:"ttl,omitempty"`
+	IsolationSegment     string `json:"isolation_segment"`
+	TerminateFrontendTLS bool   `gorm:"default:false" json:"terminate_frontend_tls,omitempty"`
+	ALPNs                string `json:"alpns,omitempty"`
```

Comment:
I have changed the property name to be ALPNs as it's a list of ALPNs that were getting stored.

Reply:
1. It's a csv. I will add that comment. I was having trouble with a uniqueness function when unmarshalling to a String Slice:

```
type StringSlice []string

func (ss *StringSlice) MarshalJSON() ([]byte, error) {
	slices.Sort(*ss)
	return json.Marshal(ss)
}

func (ss *StringSlice) UnmarshalJSON(b []byte) error {
	var s string
	if err := json.Unmarshal(b, &s); err != nil {
		return err
	}
	*ss = strings.Split(s, ",")
	return nil
}

func (ss *StringSlice) Matches(other *StringSlice) bool {
	return slices.Equal(*other, *ss)
}
```

2. I don't think we need to add the new properties to the uniqueness index